granulate_utils/metrics/sampler.py

```diff
@@ -65,21 +65,13 @@ def __init__(
         self._hostname = hostname
         self._applications_metrics = applications_metrics
         self._spark_samplers: List[Collector] = []
+        self._master_address: Optional[str] = None
+        self._cluster_mode: Optional[str] = None
 
         if (cluster_mode is not None) and (master_address is not None):
```

Comment:
If both must be given, it's an error to give only one. `assert` here? Otherwise, your function accepts arguments that it silently ignores.

Reply:
e43731f86d25e2832dfba862812b2d96ba8db67a